examples/example_hyper.rs

```diff
@@ -13,72 +13,78 @@
-    r.register(Box::new(gauge.clone())).unwrap();
-
-    counter.inc();
-    assert_eq!(counter.get() as u64, 1);
-    counter.inc_by(42.0).unwrap();
-    assert_eq!(counter.get() as u64, 43);
-
-    let c2 = counter.clone();
-    thread::spawn(move || {
-        loop {
-            thread::sleep(Duration::from_millis(300));
-            c2.inc();
-        }
-    });
-
-    thread::spawn(move || {
-        loop {
-            thread::sleep(Duration::from_millis(500));
-            gauge.inc();
-
-            thread::sleep(Duration::from_millis(500));
-            gauge.dec();
-
-            thread::sleep(Duration::from_millis(500));
-            gauge.set(42.0);
-        }
-    });
+lazy_static! {
+    static ref HTTP_COUNTER: Counter = {
+        let counter_opts =
+            Opts::new("example_http_requests_total", "Total number of HTTP requests made.")
+                .const_label("handler", "all");
 
-    let encoder = TextEncoder::new();
-    // Http server
-    run("127.0.0.1:9898", r, encoder);
+        let counter = Counter::with_opts(counter_opts).unwrap();
+
+        prometheus::register(Box::new(counter.clone())).unwrap();
+
+        counter
+    };
+
+    static ref HTTP_BODY_GAUGE: Gauge = {
+        let gauge_opts =
+            Opts::new("example_http_response_size_bytes", "The HTTP response sizes in bytes.")
+                .const_label("handler", "all");
+
+        let gauge = Gauge::with_opts(gauge_opts).unwrap();
+
+        prometheus::register(Box::new(gauge.clone())).unwrap();
+
+        gauge
+    };
+
+    static ref HTTP_REQ_HISTOGRAM: Histogram = {
+        let histogram_opts =
+            HistogramOpts::new(
+                "example_http_request_duration_microseconds",
+                "The HTTP request latencies in microseconds.")
+                .const_label("handler", "all");
+
+        let histogram = Histogram::with_opts(histogram_opts).unwrap();
+
+        prometheus::register(Box::new(histogram.clone())).unwrap();
+
+        histogram
+    };
 }
 
-// run runs a http server with a Registry and a Encoder, it blocks current thread.
-pub fn run(addr: &str, registry: Registry, encoder: TextEncoder) {
+fn main() {
+    let encoder = TextEncoder::new();
+    let addr = "127.0.0.1:9898";
     println!("listening addr {:?}", addr);
     Server::http(addr)
         .unwrap()
         .handle(move |_: Request, mut res: Response| {
-            let metric_familys = registry.gather();
+            HTTP_COUNTER.inc();
+            let start = Instant::now();
+
+            let metric_familys = prometheus::gather();
             let mut buffer = vec![];
             encoder.encode(&metric_familys, &mut buffer).unwrap();
             res.headers_mut()
                 .set(ContentType(encoder.format_type().parse::<Mime>().unwrap()));
             res.send(&buffer).unwrap();
+
+            let spend = (start.elapsed().subsec_nanos() as f64) / 1e6;
+            HTTP_REQ_HISTOGRAM.observe(spend);
```

Comment:
I think it is ok, there is no `http_request_duration` before the first http request, and if it shows in the first response, it would be `0`, which is wrong.